### ISSUE TITLE
[BUG]sktime has codec error in Windows 11 Traditional Chinese System

### ISSUE THREAD

**Issue by fhopecc** [sktime/sktime#7479]
```python
from sktime.forecasting.arima import AutoARIMA
```
cause UnicodeDecodeError as follwoing:

'''
  File "C:\Users\lgzhangjian\scoop\apps\python310\3.10.11\lib\site-packages\sktime\base\_base.py", line 68
, in <module>
    from sktime.utils._estimator_html_repr import _HTMLDocumentationLinkMixin
  File "C:\Users\lgzhangjian\scoop\apps\python310\3.10.11\lib\site-packages\sktime\utils\_estimator_html_r
epr.py", line 190, in <module>
    _STYLE = style_file.read()
UnicodeDecodeError: 'cp950' codec can't decode byte 0xe2 in position 4505: illegal multibyte sequence
'''

It is because "_estimator_html_repr.css" codec is 'utf8', 
but "open" in  Windows 11 Traditional Chinese System with default codec 'cp950' as following:

with open(Path(__file__).parent / "_estimator_html_repr.css") as style_file: # default codec with cp950 in windows 11 tranditional chinese
    # use the style defined in the css file
    _STYLE = style_file.read()

it should open with specified codec with 'utf8'. 

with open(Path(__file__).parent / "_estimator_html_repr.css", encoding='utf8') as style_file: chinese
    # use the style defined in the css file
    _STYLE = style_file.read()

This fixes the problem.

sktime versions:
Name: sktime
Version: 0.34.0

**Comment by yarnabrina** [comment]
FYI @mateuszkasprowicz

@fhopecc thanks for tge bug report. I do not have a Chinese system to reproduce, but using utf-8 sounds consistent enough change. Since you have already identified the fix, do you want to contribute as well?

**Comment by mateuszkasprowicz** [comment]
Great catch @fhopecc. I think I saw a potentially related issue in the recent PyData's sktime presentation in one of the notebooks (afair Martin's). Strange display of some characters. Might be this.

**Comment by fkiraly** [comment]
this might be related to https://github.com/sktime/sktime/pull/6862 and the fix by @fnhirwa

**Comment by fnhirwa** [comment]
yeah, this can be fixed by passing the encoding parameter to the open function.
I am going to push the fix soon.
